packages/hot-updater/src/commands/build:native.ts

```diff
@@ -0,0 +1,303 @@
+} from "@/utils/fingerprint";
+import { runNativeBuild } from "@/utils/nativeBuild/runNativeBuild";
+import { printBanner } from "@/utils/printBanner";
+import { getNativeAppVersion } from "@/utils/version/getNativeAppVersion";
+import { ExecaError } from "execa";
+import picocolors from "picocolors";
+
+export interface NativeBuildOptions {
+  outputPath?: string;
+  interactive: boolean;
+  message?: string;
+  platform?: Platform;
+}
+
+export const nativeBuild = async (options: NativeBuildOptions) => {
+  printBanner();
+
+  const cwd = getCwd();
+
+  // const gitCommit = await getLatestGitCommit();
+  // const [gitCommitHash, gitMessage] = [
+  //   gitCommit?.id() ?? null,
+  //   gitCommit?.summary() ?? null,
+  // ];
+
+  const platform =
+    options.platform ??
+    (options.interactive
+      ? await getPlatform("Which platform do you want to deploy?")
+      : null);
+
+  if (p.isCancel(platform)) {
+    return;
+  }
+
+  if (!platform) {
+    p.log.error(
+      "Platform not found. -p <ios | android> or --platform <ios | android>",
+    );
+    return;
+  }
+
+  const config = await loadConfig({ platform, channel: /* todo */ "DUMMY" });
+  if (!config) {
+    console.error("No config found. Please run `hot-updater init` first.");
+    process.exit(1);
+  }
+
+  const target: {
+    appVersion: string | null;
+    fingerprintHash: string | null;
+  } = {
+    appVersion: null,
+    fingerprintHash: null,
+  };
+
+  // calculate fingerprint of the current file state in the native platform directory
+
+  if (config.updateStrategy === "fingerprint") {
+    const s = p.spinner();
+    const localFingerprint = (await readLocalFingerprint())?.[platform];
+    if (!localFingerprint) {
+      p.log.warn(
+        `Resolving fingerprint for ${platform} failed. Building native will generate it.`,
+      );
+    }
+    s.start(`Fingerprinting (${platform})`);
+
+    // generate fingerprint.json automatically
+    const generatedFingerprint = (await createAndInjectFingerprintFiles())[
+      platform
+    ];
+
+    s.stop(`Fingerprint(${platform}): ${generatedFingerprint}`);
+
+    if (!isFingerprintEquals(localFingerprint, generatedFingerprint)) {
+      p.log.info(
+        `${picocolors.blue(`fingerprint.json, ${platform} fingerprint config files`)} have been changed.`,
+      );
+    }
+    target.fingerprintHash = generatedFingerprint.hash;
+  } else if (config.updateStrategy === "appVersion") {
+    const s = p.spinner();
+    s.start(`Get native app version (${platform})`);
+
+    const appVersion = await getNativeAppVersion(platform);
+
+    s.stop(`App Version(${platform}): ${appVersion}`);
+
+    target.appVersion = appVersion;
+
+    if (!target.appVersion) {
+      p.log.error(`Failed to retrieve native app version of ${platform}`);
+      return;
+    }
+  }
+
+  // const nativeBuildConfig = config.nativeBuild;
+
+  const outputPath = options.outputPath ?? path.join(cwd, "nativebuild"); // TODO any suggestion?
```

Comment:
How about using the default path that is generated by Android (per platform)?

Reply:
or it would be cool if we configure .hot-updater/output as a common result path for bundle or build.

Reply:
` .hot-updater/output` 

it's okay.

Reply:
Can I separate this work as a other PR? @gronxb

Reply:
yes